airflow/serialization/serialized_objects.py

```diff
@@ -284,7 +286,7 @@ def _deserialize(cls, encoded_var: Any) -> Any:
         elif type_ == DAT.TIMEDELTA:
             return datetime.timedelta(seconds=var)
         elif type_ == DAT.TIMEZONE:
-            return Timezone(var)
+            return pendulum.timezone(var)
```

Comment:
For consistency with _deserilalized_dag, should this be

```suggestion
            return self._deserialize_timezone(var)
```

(I think the functionality is identical.)

Reply:
@ecerulm ^^ can you take a look at this suggestion please and rebase to the latest main at the same time, thanks